pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/kit/KitFinalSentBaseUseCase.java

```diff
@@ -49,13 +49,18 @@ protected void triggerEvents(String ddpLabel, KitRequestShipping kitRequestShipp
         }
         if (kitPayload.getDdpInstanceDto().isESUpdatePossible()) {
             try {
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping,
-                        kitPayload.getDdpInstanceDto(), "ddpLabel", "ddpLabel",
-                        ddpLabel, new PutToNestedScriptBuilder()).export();
+                //to avoid updating old deactivated record in ES because of search by label .. update only if NO de-activated record exists with same label
+                //check if de-activated kit request exists for same kit-request-id and skip ES update
+                if (kitDao.getDeactivatedKitCountByLabel(kitRequestShipping) == 0) {
```

Comment:
Other alternative is check kitId in elastic and skip updating if different kit Id. Went with check in DB